listenbrainz/webserver/errors.py

```diff
@@ -138,7 +138,7 @@ def handle_error(error, code):
                 otherwise
         """
         if current_app.config.get('IS_API_COMPAT_APP') or request.path.startswith(API_PREFIX):
-            return jsonify({'code': code, 'error': error.description}), code
+            return crossdomain(jsonify({'code': code, 'error': error.description}), code)
```

Comment:
Only pass the jsonify to crossdomain which will add headers to the response. The `handle_error` method has to return something like `(Response, code)` so code should be outside call to crossdomain.
```suggestion
            return crossdomain(jsonify({'code': code, 'error': error.description})), code
```